### PR TITLE
Add MLP

### PR DESCRIPTION
**Pull request by jwnys**
I find myself dragging along some simple MLPs, so maybe it's useful adding it to netket. Not sure how useful it is for others.

### CODE REVIEW

File: netket/models/mlp.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    output_dim: int = 1
+    """The output dimension"""
+    hidden_dims: Tuple[int] = None
+    """The size of the hidden layers"""
+    alpha_hidden_dims: Tuple[int] = None
+    """The size of the hidden layers provided as number of times the input size"""
+    param_dtype: Any = np.float64
+    """The dtype of the weights."""
+    hidden_activation: Any = nknn.gelu
+    """The nonlinear activation function after each hidden layer."""
```
Comment (gcarleo) [inline netket/models/mlp.py:28]: can we pass here a list or must it be the same activation for all layers?

**Comment by gcarleo** [comment]
I think this is very useful indeed! I would just adjust the interface for activation functions (see comments)

File: netket/models/mlp.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+from netket.utils import deprecate_dtype
+from netket.utils.types import NNInitFunc
+from netket import nn as nknn
+
+default_kernel_init = lecun_normal()
+default_bias_init = zeros
+
+
+@deprecate_dtype
```
Comment (PhilipVinc) [inline netket/models/mlp.py:16]: It's a new feature so no need to deprecate old syntax

File: netket/models/mlp.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+class MLP(nn.Module):
```
Comment (PhilipVinc) [inline netket/models/mlp.py:17]: Can you add a class do string with some reference and/or more long form explanation of the various attributes/options?

**Review by PhilipVinc — COMMENTED** [review]
Can you also add it to the generated documentation ?

File: netket/models/mlp.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    hidden_dims: Tuple[int] = None
+    """The size of the hidden layers"""
+    alpha_hidden_dims: Tuple[int] = None
+    """The size of the hidden layers provided as number of times the input size"""
+    param_dtype: Any = np.float64
+    """The dtype of the weights."""
+    hidden_activation: Any = nknn.gelu
+    """The nonlinear activation function after each hidden layer."""
+    output_activation: Any = None
+    """The nonlinear activation at the output layer."""
```
[the remark this answers is not in the document]
Comment (jwnys) [reply netket/models/mlp.py:30]: I prefer to separate output from hidden layers. It's more common to do it this way, since the output layer is typically different from the rest. For example, in MLPs for regression, the output activation is just linear, and for classification the output is typically as sigmoid or a softmax. But now you can specify also the activation function for all hidden layers separately.

**Review by PhilipVinc — COMMENTED** [review]
Can you add a bunch of tests checking
 - that squeeze output squeezes the output
 - that it throws an error if hidden activation mismatches the length of n layers

Also, do we have 1 or 2 references to cite in NQS literature that use MLP?

Finally, it's stylistic but I try to keep line size below 89 characters normally. It's not strictly enforces but when you can cut them, please do. All those docstrings are super long,

**Comment by PhilipVinc** [comment]
with those minor things fifed, it's good to go

File: netket/models/mlp.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    @nn.compact
+    def __call__(self, input):
+        hidden_dims = self.hidden_dims
+        if hidden_dims is None:
+            if self.alpha_hidden_dims is not None:
+                hidden_dims = [
+                    int(nh * input.shape[-1]) for nh in self.alpha_hidden_dims
+                ]
+        elif self.alpha_hidden_dims is not None:
+            raise ValueError(
```
[the remark this answers is not in the document]
Comment (jwnys) [reply netket/models/mlp.py]: I added some tests

File: netket/models/mlp.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+                "Cannot specify both hidden_dims and alpha_hidden_dims, choose one way to provide the hidden dimensions"
+            )
+
+        if hidden_dims is None:
+            hidden_dims = []
```
[the remark this answers is not in the document]
Comment (jwnys) [reply netket/models/mlp.py]: added a test
